hazelcast/src/main/java/com/hazelcast/cache/impl/operation/AbstractCacheOperation.java

```diff
@@ -62,8 +67,11 @@ public String getServiceName() {
     @Override
     public final void beforeRun()
             throws Exception {
-        ICacheService service = getService();
-        cache = (CacheRecordStore) service.getOrCreateRecordStore(name, getPartitionId());
+        cacheService = getService();
+        if (cacheService.wanReplicationEnabled(name)) {
```

Comment:
We don't want to create unnecessary objects after each operation run. (DefaultCacheEntryView to be exact). I'll move this check to `CacheRecordStore`.
Also we need this check as oss doesn't support wan replication.